test/Conversion/SCFToCalyx/convert_controlflow.mlir

```diff
@@ -198,23 +293,88 @@ module {
+// to be inlined into the combinational group. This also tests multiple returns.
+
+// CHECK:      module  {
+// CHECK-NEXT:   calyx.program "main"  {
+// CHECK-NEXT:     calyx.component @main(%in0: i32, %in1: i32, %in2: i32, %clk: i1 {clk}, %reset: i1 {reset}, %go: i1 {go}) -> (%out0: i32, %done: i1 {done}) {
+// CHECK-NEXT:       %true = hw.constant true
+// CHECK-NEXT:       %std_ge_0.left, %std_ge_0.right, %std_ge_0.out = calyx.std_ge "std_ge_0" : i32, i32, i1
+// CHECK-NEXT:       %std_add_1.left, %std_add_1.right, %std_add_1.out = calyx.std_add "std_add_1" : i32, i32, i32
+// CHECK-NEXT:       %std_add_0.left, %std_add_0.right, %std_add_0.out = calyx.std_add "std_add_0" : i32, i32, i32
+// CHECK-NEXT:       %ret_arg0_reg.in, %ret_arg0_reg.write_en, %ret_arg0_reg.clk, %ret_arg0_reg.reset, %ret_arg0_reg.out, %ret_arg0_reg.done = calyx.register "ret_arg0_reg" : i32, i1, i1, i1, i32, i1
+// CHECK-NEXT:       calyx.wires  {
+// CHECK-NEXT:         calyx.assign %out0 = %ret_arg0_reg.out : i32
+// CHECK-NEXT:         calyx.comb_group @bb0_2  {
+// CHECK-NEXT:           calyx.assign %std_ge_0.left = %std_add_1.out : i32
+// CHECK-NEXT:           calyx.assign %std_ge_0.right = %in2 : i32
+// CHECK-NEXT:           calyx.assign %std_add_1.left = %std_add_0.out : i32
+// CHECK-NEXT:           calyx.assign %std_add_0.left = %in0 : i32
+// CHECK-NEXT:           calyx.assign %std_add_0.right = %in1 : i32
+// CHECK-NEXT:           calyx.assign %std_add_1.right = %in1 : i32
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @ret_assign_0  {
+// CHECK-NEXT:           calyx.assign %ret_arg0_reg.in = %in1 : i32
+// CHECK-NEXT:           calyx.assign %ret_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.group_done %ret_arg0_reg.done : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @ret_assign_1  {
+// CHECK-NEXT:           calyx.assign %ret_arg0_reg.in = %in2 : i32
+// CHECK-NEXT:           calyx.assign %ret_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.group_done %ret_arg0_reg.done : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:       }
+// CHECK-NEXT:       calyx.control  {
+// CHECK-NEXT:         calyx.seq  {
+// CHECK-NEXT:           calyx.if %std_ge_0.out with @bb0_2  {
+// CHECK-NEXT:             calyx.seq  {
+// CHECK-NEXT:               calyx.enable @ret_assign_0
+// CHECK-NEXT:             }
+// CHECK-NEXT:           } else  {
+// CHECK-NEXT:             calyx.seq  {
+// CHECK-NEXT:               calyx.enable @ret_assign_1
+// CHECK-NEXT:             }
+// CHECK-NEXT:           }
+// CHECK-NEXT:         }
+// CHECK-NEXT:       }
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+// CHECK-NEXT: }
+module {
+  func @main(%a0 : i32, %a1 : i32, %a2 : i32) -> i32 {
+    %0 = addi %a0, %a1 : i32
+    %1 = addi %0, %a1 : i32
+    %b = cmpi uge, %1, %a2 : i32
+    cond_br %b, ^bb1, ^bb2
+  ^bb1:
+    return %a1 : i32
+  ^bb2:
+    return %a2 : i32
```

Comment:
I don't recall; is there a test case where there are multiple non-disjoint returns as well?

Reply:
Maybe i'm misunderstanding, but wouldn't multiple return statements always be disjoint?

Reply:
Is it possible to return multiple values, e.g. `return %a1, %a2 : i32, i32` ? If not, belay my last :-)

Reply:
Ahh, now i understand - i'll add a test.